jwst/tweakreg/astrometric_utils.py

```diff
@@ -0,0 +1,179 @@
+import os
+import requests
+
+from astropy import table
+from astropy.table import Table
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+
+from resample import resample_utils
+
+ASTROMETRIC_CAT_ENVVAR = "ASTROMETRIC_CATALOG_URL"
+DEF_CAT_URL = 'http://gsss.stsci.edu/webservices'
+
+if ASTROMETRIC_CAT_ENVVAR in os.environ:
+    SERVICELOCATION = os.environ[ASTROMETRIC_CAT_ENVVAR]
+else:
+    SERVICELOCATION = DEF_CAT_URL
+
+"""
+
+Primary function for creating an astrometric reference catalog.
+
+"""
+
+
+def create_astrometric_catalog(input_models, catalog="GAIADR2", output="ref_cat.ecsv",
+                               gaia_only=False, table_format="ascii.ecsv",
+                               existing_wcs=None, num_sources=None):
+    """Create an astrometric catalog that covers the inputs' field-of-view.
```

Comment:
Is this the field-of-view that is the union of all inputs?

Reply:
This function determines the full field-of-view for all the input models based on the available WCS information, and requests all the GAIA sources that fall in that field-of-view from the web service.  So, yes, it is intended to be the union of all the inputs.